packages/TogglClient-Core.package/TogglClient.class/instance/startTimer.st

```diff
@@ -1,7 +1,7 @@
 actions
 startTimer 
 
-	self timer isActive ifFalse: [
+	self timerIsRunning ifFalse: [
```

Comment:
why do we have a timerIsRunning function now?

Reply:
Der StartStopButton muss sein Label anpassen danach, ob der Timer gerade läuft oder nicht. Der ButtonSpec bietet dafür eine changeLabelWhen: Observer(?) Funktion an. Funktionskommentar:
"When the button handles aSymbol in its update: handler, treat it
	as notification that the button label should be updated."

Ich habe das jetzt so verwendet:
		label: #startStopTimerLabel;
		changeLabelWhen: #timerIsRunning;
		action: #actionToggleStartStop;
Also wenn timerIsRunning sich ändert, wird startStopTimerLabel gerufen, das je nach timerIsRunning oder nicht ein anderes Label erhält. 

Dafür habe ich noch in 

actionToggleStartStop

	self timerIsRunning
		ifTrue: [self stopTimerAndAddTimeEntry]
	        ifFalse: [self startTimer].
      self changed: #timerIsRunning

ein changed hinzugefügt. 

Keine Ahnung, ob das so der sinnvollste Weg ist. Man könnte genauso gut bei actionToggleStartStop ein self changed: #startStopTimerLabel hinzufügen vermtl, das ist aber semantisch nicht so schön - weil das was sich ja eig. ändert ist, ob der Timer läuft.

Reply:
Und timer isActive zu isTimerRunning zu kapseln finde ich gar nicht so schlecht. (Falls wir mal theoretisch die zugrundelegen Datenstruktur der Stopwatch gegen was anderes austauschen sollten, müssten wir das dann nur noch in isTimerRunning ändern)

Reply:
Okay find ich fair